PRINCIPLES.md

```diff
@@ -0,0 +1,259 @@
+
+Together, the aggregate of all configuration data for a system form its "Desired State". The "Desired State" of a system is defined as data sufficient to recreate the system from nothing so that instances of the system are behaviorally indinstinguishable.
+
+#### Why must the Desired State be declarative data?
+
+This is a subtle but important point.
+In deference to the work done by the Infrastructure as Code community (IaC), we believe that this was the intent of that movement to begin with.
+However, we have in practice seen a misunderstanding in this area, and many implementations have considered imperative scripts or programatic definitions for provisioning infrastructure to be a sufficient implementation of Infrastructure as Code. We disagree.
+
+We make the distinction explicitly for two important reasons:
+
+Firstly, it forces a separation of concerns between the Desired State (_what_ a system is) and _how_ the system is made to reach that state.
+This modular approach enables the implementation details of operations (the _how_) to be separated and iterated on independently from the system configuration (the _what_). 
+It also enables different tools and implementations to use the same Desired State declaration and interoperate against a common data language. 
+These modular systems are more flexible.
+For example, a Python program could verify a configuration file, while a C++ executable actually implements the declaration into a running system. Encoding the Desired State with a programming language ties the implementation to it, or forces other components to create a fully featured interpreter.
+
+Secondly, verifying the correctness and self-consistency of data is significantly less complex than verifying the correctness of a program's behaviour, which is fundamentally undecidable. 
+Verifying that a set of declarations is correct, however, _is_ decidable, even if sometimes computationally expensive.
+As a general rule, this implies that the data language used to defined the Desired State should have no control flow structure, and consist exclusively of referentially transparent expressions without side effects. 
+In other words, a human-readable data-serialization format.
+
+It is also preferrable to use a widely supported language to define the Desired State for the sake of interoperability. For example, YAML, XML, JSON and TOML all have broad support and well defined specifications, although there are many other suitable candidates.
+
+#### Why is human readability required?
+
+Operationally, it has been proven time and time again that the canonical Desired State of a system should be human-readable and writable. (See _The UNIX Philosophy (1995) by Mike Gancarz_ and _The Pragmatic Programmer (2000) Section 14 "The Power of Plain Text" by Andrew Hunt and David Thomas_)
```

Comment:
Proven is a strong word, and I think we should wait until there is academic research that proves this 
```suggestion
The consensus among operators that the desired state of a system should be human-readable and writable is continuing to grow.
```

Reply:
@moshloop Are you saying you find these resources to be not credible?

Is this sufficient?

```diff
- Operationally, it has been proven time and time again…
+ Operationally, consensus continues to grow that…
```

Reply:
This is not true of ML systems. Human understanding or even observability is an area of ongoing research. Definitely not consensus yet. Growing, maybe. Desirable yes.